libs/src/services/identity/IdentityService.ts

```diff
@@ -141,6 +141,24 @@ export class IdentityService {
     })
   }
 
+  async getUserEmail() {
+    const headers: {
+      [AuthHeaders.MESSAGE]?: string
+      [AuthHeaders.SIGNATURE]?: string
+    } = {}
+    const signDataResult = await this._signData()
+    if (signDataResult) {
```

Comment:
would it make sense for _signData to return an object that is compatible with headers? so like its return type is 
```
{
   [AuthHeaders.MESSAGE]?: string
   [AuthHeaders.SIGNATURE]?: string
}
```
and then we can just do `headers = signDataResult` ?

Reply:
Agreed, if signedDataResult is not returned, it shouldn't attempt to make the request

Reply:
ahh thats also a good call ^